public/app/features/templating/template_srv.ts

```diff
@@ -291,63 +301,96 @@ export class TemplateSrv implements BaseTemplateSrv {
 
     this.regex.lastIndex = 0;
 
-    return target.replace(this.regex, (match, var1, var2, fmt2, var3, fieldPath, fmt3) => {
-      const variableName = var1 || var2 || var3;
-      const variable = this.getVariableAtIndex(variableName);
-      let fmt = fmt2 || fmt3 || format;
-
-      if (scopedVars) {
-        const value = this.getVariableValue(variableName, fieldPath, scopedVars);
-        const text = this.getVariableText(variableName, value, scopedVars);
+    // If we get passed this interpolations map we will also record all the expressions that were replaced
+    if (interpolations) {
+      return this._replaceInVariableRegex(target, format, (match, variableName, fieldPath, fmt) => {
+        const expr = `${variableName}${fieldPath ? `.${fieldPath}` : ''}`;
+        const result = this.evaluateVariableExpression(match, variableName, fieldPath, fmt, scopedVars);
+        interpolations.set(expr, result === match ? null : result);
+        return result;
+      });
+    }
 
-        if (value !== null && value !== undefined) {
-          if (scopedVars[variableName].skipFormat) {
-            fmt = undefined;
-          }
+    return this._replaceInVariableRegex(target, format, (match, variableName, fieldPath, fmt) => {
+      return this.evaluateVariableExpression(match, variableName, fieldPath, fmt, scopedVars);
+    });
```

Comment:
@gelicia, in your branch (https://github.com/grafana/grafana/compare/kristina/correlations-log-line?expand=1) it's slightly different with evaluation happening twice. Do we need it for #61782?

Reply:
You can just merge this, it's most likely that I missed something and I can figure out what we need.